packages/braid-design-system/src/lib/components/Autosuggest/Autosuggest.docs.tsx

```diff
@@ -482,6 +464,130 @@ const docs: ComponentDocs = {
+            portions matching the input value. You can use the{' '}
+            <Strong>automaticHighlights</Strong> prop to highlight either the{' '}
+            <Strong>matching</Strong> or <Strong>remaining</Strong> portion of
+            each suggestion.
+          </Text>
+        </>
+      ),
+      Example: ({ id, setDefaultState, setState, getState }) =>
+        source(
+          <>
+            {setDefaultState('textfield', 'App')}
+            {setDefaultState('suggestion', 'Apples')}
+
+            <Stack space="large">
+              <TextField
+                label="Label"
+                id={id}
+                onChange={setState('textfield')}
+                value={getState('textfield')}
+              />
+              <Columns space="gutter">
+                {['matching', 'remaining'].map((highlightType) => (
+                  <Column key={highlightType}>
+                    <Stack space="small">
+                      <Text size="small" tone="secondary">
+                        Highlight <Strong>{highlightType}</Strong>
+                      </Text>
+                      <Inline space="none">
+                        {parseHighlights(
+                          getState('suggestion'),
+                          highlightSuggestions(
+                            getState('suggestion'),
+                            getState('textfield'),
+                            highlightType === 'matching'
+                              ? 'matching'
+                              : 'remaining',
+                          ).map(({ start, end }) => [start, end]),
+                        ).map((part, index) => (
+                          <Text
+                            key={index}
+                            weight={part.highlight ? 'strong' : 'regular'}
+                          >
+                            {part.text}
+                          </Text>
+                        ))}
+                      </Inline>
+                    </Stack>
+                  </Column>
+                ))}
+              </Columns>
+            </Stack>
+          </>,
+        ),
+      code: false,
+    },
+
+    {
+      description: (
+        <>
+          <Text>
+            If <Strong>automaticHighlights</Strong> is not suitable for your use
+            case, you can provide explicit highlight ranges for each suggestion.
+          </Text>
+        </>
+      ),
+      Example: ({ id }) =>
+        source(
+          <Autosuggest
+            label="Label"
+            id={`${id}_highlights`}
+            value={{ text: 'App' }}
+            onChange={() => {}}
+            suggestions={[
+              {
+                text: 'Apples',
+                value: 1,
+                highlights: [{ start: 2, end: 6 }],
+              },
+              {
+                text: 'Bananas',
+                value: 2,
+                highlights: [{ start: 0, end: 3 }],
+              },
+            ]}
+          />,
+        ),
```

Comment:
Wanted to keep this example relatively simple so the code snippet is easy to understand. However, it requires the user to click on the input field to see the highlights so open to a suggestion on improving that